en/tutorials/tutorial_hello_sky.md

```diff
@@ -1,135 +1,177 @@
+* Rename (or delete) the **px4_simple_app** directory. 
+
+## Minimal Application
+
+In this section we create a *minimal application* that just prints out `Hello Sky!`. This consists of a single c file and a *cmake* definition (which tells the toolchain how to build the application). 
+
+1. Create a new directory **Firmware/src/examples/px4_simple_app**.
+1. Create a new C file in that directory named **px4_simple_app.c**. Copy in the text below.
+
+   > **Tip** Note the default header and code style in this file - all contributions to PX4 should adhere to it.
+
+   ```c
+   /****************************************************************************
+    *
+    *   Copyright (c) 2012-2016 PX4 Development Team. All rights reserved.
+    *
+    * Redistribution and use in source and binary forms, with or without
+    * modification, are permitted provided that the following conditions
+    * are met:
+    *
+    * 1. Redistributions of source code must retain the above copyright
+    *    notice, this list of conditions and the following disclaimer.
+    * 2. Redistributions in binary form must reproduce the above copyright
+    *    notice, this list of conditions and the following disclaimer in
+    *    the documentation and/or other materials provided with the
+    *    distribution.
+    * 3. Neither the name PX4 nor the names of its contributors may be
+    *    used to endorse or promote products derived from this software
+    *    without specific prior written permission.
+    *
+    * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+    * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+    * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    * POSSIBILITY OF SUCH DAMAGE.
+    *
+    ****************************************************************************/
+
+   /**
+    * @file px4_simple_app.c
+    * Minimal application example for PX4 autopilot
+    *
+    * @author Example User <mail@example.com>
+    */
+
+   #include <px4_config.h>
+   #include <px4_tasks.h>
+   #include <px4_posix.h>
+   #include <unistd.h>
+   #include <stdio.h>
+   #include <poll.h>
+   #include <string.h>
+
+   #include <uORB/uORB.h>
+   #include <uORB/topics/sensor_combined.h>
+   #include <uORB/topics/vehicle_attitude.h>
+
+   __EXPORT int px4_simple_app_main(int argc, char *argv[]);
```

Comment:
For the entry point, what here is "necessary"? I assume you have to always:
1. EXPORT the main function.
2. That the main function must always have that signature.
Anything else?

What is PX4_INFO? Standard output to the PX4 or Nuttshell? Are there other variants? ie.g. PX4_WARNING?

Reply:
Thanks. I've made this part of the example truly minimal and added these notes. I think it is now easier to work out what is actually important.